pkgs/development/tools/mold/default.nix

```diff
@@ -0,0 +1,35 @@
+{ stdenv, fetchFromGitHub, lib, autoPatchelfHook, cmake, tbb, llvmPackages_latest, xxHash, zlib, openssl }:
+stdenv.mkDerivation rec {
+  pname = "mold";
+  version = "0.9.1";
+
+  src = fetchFromGitHub {
+    owner = "rui314";
+    repo = "mold";
+    rev = "v${version}";
+    sha256 = "sha256-yIkW6OCXhlHZ1jC8/yMAdJbSgY9K40POT2zWv6wYr5E=";
+  };
+
+  buildInputs = [ tbb zlib openssl ];
+  nativeBuildInputs = [ autoPatchelfHook cmake xxHash ];
+
+  dontUseCmakeConfigure = true;
+  EXTRA_LDFLAGS = "--ld-path=${llvmPackages_latest.lld}/bin/ld.lld";
+  LTO = 1;
+  makeFlags = [ "PREFIX=${placeholder "out"}" ];
+
+  preInstall = ''
+    mkdir -p $out $out/bin $out/share/man/man1
+  '';
+
+  postInstall = ''
+    ln -s "$out/lib/mold/mold-wrapper.so" "$out/bin/mold-wrapper.so";
```

Comment:
Why do we want a link to bin? It is a shared object, not a binary, right?

Reply:
@SCOTT-HAMILTON said:

> You should also probably add a symbolic link of $out/lib/mold/mold-wrapper.so to $out/bin because of https://github.com/rui314/mold/blob/9a09c777d9460ebe7eb498d1cc0327915f8bbcdc/subprocess.cc#L271

Reply:
The bin directory is for executables that you can run in your shell. I'd would rather see this path patched instead.

Reply:
You can remove a symlink in 0.9.2, as mold installed as $out/bin/mold checks if $out/lib/mold-wrapper.so exists whatever $bin is. But 0.9.1 unfortunately tries to do that only if $bin is /usr or /usr/local. So, I think we need it for now.

Reply:
You can also fetch the commit it was changed as a patch.

Reply:
I updated this PR to use 0.9.2, released in the last few hours, and deleted this symlink.